doc/lint-lifecycle.md

```diff
@@ -50,17 +50,30 @@ In general removal is preceded by a period of deprecation.
 
 ## State Transitions
 
+### Implementation
+
+Once a lint proposal is accepted, it is safe to implement. A change that lands a new lint should
+have a corresponding `CHANGELOG` entry.
+
 ### Deprecation
 
 Implemented lints can be deprecated.
 
 Deprecating lints that are in common lint sets (e.g., in [package:lints](https://github.com/dart-lang/lints)
 can be impactful so should be done with care.
 
+(**NOTE**: A change that deprecates an existing lint should have a corresponding `CHANGELOG` entry.)
+
 ### Marking Stable
 
 Experimental lints should aspire to be stable. An experimental lint is a candidate for stable when it has
 
 * complete semantics
 * complete implementation (with no false-positives)
 * established long-term value (e.g., inclusion in a recommended lint set)
+
+### Removal
+
+Implemented lints can be removed.
```

Comment:
We might want to be more specific about which states can be transitioned to which other states. In particular, "Implemented" isn't very informative because it isn't a state. We might, for example, want to require that only deprecated or experimental lints can be deleted in order to ensure that users have some form of notification.